docs/plugins/spectate.rst

```diff
@@ -2,16 +2,50 @@ spectate
 ========
 
 .. dfhack-tool::
-    :summary: Automatically follow exciting dwarves.
+    :summary: Automatically follow productive dwarves.
     :tags: fort interface
-    :no-command:
 
 Usage
 -----
 
 ::
 
     enable spectate
+    disable spectate
+    spectate <option> <value>
 
-The plugin will automatically switch which dwarf is being followed periodically,
-preferring dwarves on z-levels with the highest job activity.
+
+When enabled, the plugin will automatically switch which dwarf is being
+followed periodically, preferring dwarves on z-levels with the highest
+job activity.
+
+To set features that toggle between two states, use {0,1} to specify
```

Comment:
could we use words instead of numbers to enable and disable? Maybe something similar to https://docs.dfhack.org/en/stable/docs/tools/dwarfmonitor.html#dwarfmonitor and/or https://docs.dfhack.org/en/stable/docs/tools/buildingplan.html#buildingplan?

e.g.

```
Usage
-------

::

    enable spectate
    spectate
    spectate enable|disable <feature>
    spectate set <setting> <value>
```

Reply:
So, that would look like this. (it's more code than what I have currently)
Making this change will require testing, and reformatting the docs to distinguish "features" and "settings" I don't think it's worth the effort
```cpp
        auto get_feature = [](std::string &arg){
            if(arg == "auto-unpause"){
                return UNPAUSE;
            } else if (arg == "auto-disengage") {
                return DISENGAGE;
            } else if (arg == "focus-jobs") {
                return JOB_FOCUS;
            } else if (arg == "tick-threshold") {
                return TICK_THRESHOLD;
            }
            return INVALID;
        };
        if (parameters.size() >= 2 && parameters.size() <= 3) {
            bool state;
            bool set = false;
            if (parameters[0] == "enable") {
                state = true;
            } else if (parameters[0] == "disable") {
                state = false;
            } else if (parameters[0] == "set") {
                set = true;
            } else {
                return DFHack::CR_WRONG_USAGE;
            }
            switch (get_feature(parameters[1])) {
                case UNPAUSE:
                    if (set) {
                        return CR_WRONG_USAGE;
                    }
                    enable_auto_unpause(out, state);
                    break;
                case DISENGAGE:
                    if (set) {
                        return CR_WRONG_USAGE;
                    }
                    disengage_enabled = state;
                    break;
                case JOB_FOCUS:
                    if (set) {
                        return CR_WRONG_USAGE;
                    }
                    focus_jobs_enabled = state;
                    break;
                case TICK_THRESHOLD:
                    if (!set) {
                        return CR_WRONG_USAGE;
                    }
                    try {
                        tick_threshold = std::stol(parameters[2]);
                    } catch (const std::exception &e) {
                        out.printerr("%s\n", e.what());
                    }
                    break;
                case INVALID:
                    return DFHack::CR_WRONG_USAGE;
            }
        }
```